### PR TITLE
API for creating cloud volume backup

### PR DESCRIPTION
**Pull request by jeffibm**
Creating an API end point for cloud volume backup

API initiated from https://github.com/ManageIQ/manageiq-ui-classic/pull/8187

@miq-bot add-reviewer @Fryguy 
@miq-bot add-label enhancement
@miq-bot assign @Fryguy

### CODE REVIEW

**Comment by kavyanekkalapu** [comment]
@jeffibm Could you api endpoint response? Example-> https://github.com/ManageIQ/manageiq-api/pull/1006#issuecomment-779993161

File: app/controllers/api/cloud_volumes_controller.rb [hunk 1]
```diff
@@ -41,5 +41,12 @@ def options
         super
       end
     end
+
+    def create_backup_resource(type, id, data)
+      cloud_volume = resource_search(id, type)
+      {:task_id => cloud_volume.backup_create_queue(User.current_userid, data)}
+    rescue => err
+      action_result(false, err.to_s)
```
[the remark this answers is not in the document]
Comment (kbrock) [reply app/controllers/api/cloud_volumes_controller.rb:49]: api_resource is our friend. it rescues and displays the appropriate errors for us.

File: config/api.yml [hunk 2]
```diff
@@ -795,6 +795,8 @@
         :identifier: cloud_volume_delete
       - :name: safe_delete
         :identifier: cloud_volume_safe_delete
+      - :name: create_backup
+        :identifier: cloud_volume_backup
```
[the remark this answers is not in the document]
Comment (kbrock) [reply config/api.yml:799]: that is the name that we use in the model. seems that is a good plan.

File: spec/requests/cloud_volumes_spec.rb [hunk 3]
```diff
@@ -193,4 +193,20 @@
...
+      provider = FactoryBot.create(:ems_autosde, :zone => zone)
+      cloud_volume = FactoryBot.create(:cloud_volume_autosde, :ext_management_system => provider)
+
+      api_basic_authorize(action_identifier(:cloud_volumes, :create_backup, :resource_actions, :post))
+      stub_supports(cloud_volume.class, :create_backup)
+
+      payload = {:action => "create_backup", :resources => {:backup_name => "stud_backup_name"}}
+      post(api_cloud_volume_url(nil, cloud_volume), :params => payload)
+
+      expect(response).to have_http_status(:ok)
```
Comment (jrafanie) [inline spec/requests/cloud_volumes_spec.rb:209]: Do we test the failure cases?  There was a rescue clause above, if we expect failures, we should probably test how we deal with failures such as "missing permissions/identifier" or invalid payload/request.

Comment (kbrock) [reply spec/requests/cloud_volumes_spec.rb:209]: yea. the original code was not checking `supports`, so that would have been caught by an unhappy path.

**Comment by jeffibm** [comment]
```
post /api/cloud_volumes/1
{
    "action": "create_backup",
     "resource": {
         "backup_name": "stud_backup_name"
     }
}

{
    "results": [
        {
            "success": true,
            "message": "Creating Backup for Cloud Volume id: 1 name: 'e432b0c9-4c49-41e6-ba81-015d8f085576'",
            "task_id": "436037",
            "task_href": "http://localhost:3000/api/tasks/436037",
            "href": "http://localhost:3000/api/cloud_volumes/1"
        }
    ]
}
```

File: spec/requests/cloud_volumes_spec.rb (hunk 3 above)
Comment (kbrock) [reply spec/requests/cloud_volumes_spec.rb:209]: Please add 2 specs.
one with a blank `api_basic_authorize` (e.g. one on line 46 w/ rejects delete) and one with `stub_supports_not` for probably a `have_http_status(:not_found)`

**Pull request merged**